[FSAL] Consildate FSAL messages

Manimaran-MM opened the pull request:
New pipeline line workflow to consolidate all the FSAL runs to one and reporting also turns out to be one

**Workflow**: 
GerrtiHub new patch set created/recheck fsal -> trigger-fsal-on-new-patch job gets triggered -> -> all the 4 existing fsal jobs invoked with required param -> coolects result from all the 4 jobs -> Publishes summarised review comments based on the result and notifies incase of failure. Also comments will have info about the result and log path for review

**Screenshot**
<img width="886" height="392" alt="image" src="https://github.com/user-attachments/assets/74c0d919-09d8-4ee9-86cc-c2d0c67362ca" />

**Post-Merge Changes:**

- Below config change for all the existing fsal-*
   - Update new Param GERRIT_PATCHSET_REVISION and defaults to empty (This is needed to run the test on correct patch revision)
   - Update fsal-build.sh with the updated content
   - Handle the archive artifactory for result_message.txt (This artifact is used to collect the job results)

Manimaran-MM committed:
New pipeline line workflow to consolidate all the FSAL runs to one and reporting also turns out to be one

Workflow: GerrtiHub new patch set created/recheck fsal -> trigger-fsal-on-new-patch job gets triggered -> -> all the 4 existing fsal jobs invoked with required param -> coolects result from all the 4 jobs -> Publishes summarised review comments based on the result and notifies incase of failure.
Also comments will have info about the result and log path for review

Improvements: Modified "git clone" mechanism in all 4 FSAL jobs, which now saves approx 5 min per job

Note: This is changed to non-voting job

Signed-off-by: Manimaran M <manim@redhat.com>

kalebskeithley reviewed (approved):
LGTM

psathyan reviewed (commented):
/lgtm

Thanks @Manimaran-MM for addressing multiple posts in gerrit. The build summary looks good.

Could you include deprecation of https://github.com/nfs-ganesha/ci-tests/blob/centos-ci/jobs/fsal_template.yml  workflows and disable the trigger there?

Let include the deprecation as a separate commit in this pull request. 

Once the PR is merged, let's monitor the queue.

Manimaran-MM committed:
ci: deprecate fsal_template.yml workflow

Marked jobs/fsal_template.yml as DEPRECATED and added a note pointing
users to the replacement workflow: jobs/trigger-fsal-on-new-patch.yml.

This workflow is no longer in use and should not be maintained going forward.

Signed-off-by: Manimaran M <manim@redhat.com>

Manimaran-MM commented:
> /lgtm
> 
> Thanks @Manimaran-MM for addressing multiple posts in gerrit. The build summary looks good.
> 
> Could you include deprecation of https://github.com/nfs-ganesha/ci-tests/blob/centos-ci/jobs/fsal_template.yml workflows and disable the trigger there?
> 
> Let include the deprecation as a separate commit in this pull request.
> 
> Once the PR is merged, let's monitor the queue.

Done

ffilz commented:
One comment on this - can we replace FSAL_GLUSTER (that we no longer really support) with FSAL_VFS?

Manimaran-MM commented:
> One comment on this - can we replace FSAL_GLUSTER (that we no longer really support) with FSAL_VFS?

Noted @ffilz 
Will push in the next PR
